src/useLogin.tsx

```diff
@@ -4,13 +4,17 @@ import Utility from './helpers/utility'
 import UserService from './services/user.service'
 import {ExtendedError, LoggedInUserData, LoginMethod} from './types/types'
 
+export type UsernameAndPasswordLogin = {
+  isLoading: boolean
+  errorMessage: string | undefined
+  onSubmit: (username: string, password: string) => void
+}
```

Comment:
Simplify prop drilling by grouping the variables for the UsernameAndPasswordForm in the same object